src/Logging.chpl

```diff
@@ -0,0 +1,88 @@
+module Logging {
+    use Set;
+    use IO;
+    use DateTime;
+
+    /*
+     * The LogLevel enum is used to provide a strongly-typed means of
+     * configuring the logging level of a Logger object
+     */
+    enum LogLevel {DEBUG,INFO,WARN,ERROR,CRITICAL};
+    
+    /*
+     * The Logger class provides structured log messages at various levels
+     * of logging sensitivity analogous to other languages such as Python.
+     */
+    class Logger {
+        var level = LogLevel.WARN;
+        
+        var warnLevels = new set(LogLevel,[LogLevel.WARN, LogLevel.INFO,
+                           LogLevel.DEBUG]);
+        
+        var criticalLevels =  new set(LogLevel, [LogLevel.CRITICAL, LogLevel.WARN,
+                                         LogLevel.INFO,LogLevel.DEBUG]);
+                                         
+        var errorLevels =  new set(LogLevel,[LogLevel.ERROR, LogLevel.CRITICAL,
+                           LogLevel.WARN, LogLevel.INFO, LogLevel.DEBUG]);
+                             
+        var infoLevels = new set(LogLevel,[LogLevel.INFO,LogLevel.DEBUG]);  
```

Comment:
Micro-optimization, but it would be better to declare these sets with `parSafe=false` to avoid locking when accessing them. I also think they could just be created once at the module level instead of once per instance. I think they could also be `const`